tests/common/fixtures/elasticsearch.py

```diff
@@ -17,10 +18,19 @@ def es_client(delete_all_elasticsearch_documents):
     return _es_client()
 
 
+@pytest.fixture
+def es_connect():
+    # TODO handle deleting things out of this connection's index as
+    # the `es_client` fixture does
+    search.connect(hosts=[ELASTICSEARCH_URL])
+
+
 @pytest.fixture(scope="session", autouse=True)
 def init_elasticsearch(request):
-    """Initialize the test Elasticsearch index once per test session."""
+    """Initialize the test (old) Elasticsearch index once per test session."""
     client = _es_client()
+    """Connect to the newer v6.x instance of Elasticsearch once per test session"""
+    es_connect()
```

Comment:
One thing I note for both the "old" connection and also my new one is that there is no teardown for these connections (the index of the old ES is cleared out but the actual connection doesn't get killed AFAICT). It would be straightforward to add a `finalizer` that could invoke `elasticsearch_dsl.connections.remove_connection` and/or create a `disconnect` method in `h.search.connection` if that could ostensibly be useful.

Reply:
Use yield for this. There was an example of this present in the tests on my original PR. Create the connection, yield the connection then after yield tear down the connection.